docs/source-fabric/advanced/model_parallel/index.rst

```diff
@@ -0,0 +1,162 @@
+
+
+----
+
+
+**************************
+What is Model Parallelism?
+**************************
+
+There are different types of model parallelism, each with its own trade-offs.
+
+**Fully Sharded Data Parallelism (FSDP)** shards both model parameters and optimizer states across multiple GPUs, significantly reducing memory usage per GPU.
+This method, while highly memory-efficient, involves frequent synchronization between GPUs, introducing communication overhead and complexity in implementation.
+FSDP is advantageous when memory constraints are the primary issue, provided there are high-bandwidth interconnects to minimize latency.
+
+**Tensor Parallelism (TP)** splits individual tensors across GPUs, enabling fine-grained distribution of computation and memory.
+It scales well to a large number of GPUs but requires synchronization of tensor slices after each operation, which adds communication overhead.
+TP is most effective with models that have many linear layers (LLMs), offering a balance between memory distribution and computational efficiency.
+
+**Pipeline Parallelism (PP)** divides model layers into segments, each processed by different GPUs, reducing memory load per GPU and minimizing inter-GPU communication to pipeline stage boundaries.
+While this reduces communication overhead, it can introduce pipeline bubbles where some GPUs idle, leading to potential inefficiencies.
+PP is ideal for deep models with sequential architectures (LLMs), though it requires careful management to minimize idle times.
+
+Choosing a model parallelism style involves considering model architecture, hardware interconnects, and training efficiency.
+In practice, hybrid approaches combining FSDP, TP, and PP are often used to leverage the strengths of each method while mitigating their weaknesses.
+
+
+----
+
+
+***********
+Get started
+***********
+
+.. raw:: html
+
+    <div class="display-card-container">
+        <div class="row">
+
+.. displayitem::
+    :header: Fully-Sharded Data Parallel (FSDP)
+    :description: Get started training large multi-billion parameter models with minimal code changes
+    :col_css: col-md-4
+    :button_link: fsdp.html
+    :height: 180
+    :tag: advanced
+
+.. displayitem::
+    :header: Tensor Parallel (TP)
+    :description: Learn the principles behind tensor parallelism and how to apply it to your model
+    :col_css: col-md-4
+    :button_link: tp.html
+    :height: 180
+    :tag: advanced
+
+.. displayitem::
+    :header: 2D Parallel (FSDP + TP)
+    :description: Combine Tensor Parallelism with FSDP (2D Parallel) to train efficiently on 100s of GPUs
+    :button_link: tp_fsdp.html
+    :col_css: col-md-4
+    :height: 180
+    :tag: advanced
+
+.. displayitem::
+    :header: Pipeline Parallelism
+    :description: Coming soon
+    :button_link:
+    :col_css: col-md-4
+    :height: 180
+    :tag: advanced
+
+.. raw:: html
+
+        </div>
+    </div>
+
+
+----
+
+
+*********************
+Parallelisms compared
```

Comment:
I initially wrote this first, but now I'm not so sure anymore if it's the best way to present it. 
What do you like more? The text intro at the top or the explicit pros/cons bullet points here?

Reply:
I think the text reads very well, I would not remove it. However, optionally a table or section with short pro/con bullet points can be nice to summarize the text. But the text is nice because it also explains more. Maybe you could add the bullet points below each text section and make them shorter.

Reply:
I love the section personally, it's clear and factual.